packages/@aws-cdk/service-spec-importers/test/property-conversion.test.ts

```diff
@@ -232,6 +232,170 @@ test('read required properties from allOf/anyOf', () => {
   expect(requiredProps).toContain('InBoth');
 });
 
+test('oneOf containing a list of "required" properties and a required property', async () => {
+  importCloudFormationRegistryResource({
+    db,
+    report,
+    resource: {
+      typeName: 'AWS::OneOf::Required',
+      description: 'Resource Type Description',
+      definitions: {
+        OneConfiguration: {
+          type: 'object',
+          properties: {
+            Foo: { type: 'string' },
+          },
+        },
+        AnotherConfiguration: {
+          type: 'object',
+          properties: {
+            Bar: { type: 'string' },
+          },
+        },
+        SomeConfigurationProp: {
+          type: 'object',
+          description: 'does something useful',
+          properties: {
+            Type: {
+              $ref: '#/definitions/SomeType',
+            },
+            OneConfiguration: {
+              $ref: '#/definitions/OneConfiguration',
+            },
+            AnotherConfiguration: {
+              $ref: '#/definitions/AnotherConfiguration',
+            },
+          },
+          required: ['Type'],
+          oneOf: [{ required: ['OneConfiguration'] }, { required: ['AnotherConfiguration'] }],
+          additionalProperties: false,
+        },
+        SomeType: {
+          type: 'string',
+          description: 'which config to use',
+          enum: ['ONE', 'ANOTHERONE'],
+        },
+      },
+      properties: {
+        DataSourceConfiguration: {
+          $ref: '#/definitions/SomeConfigurationProp',
+        },
+      },
+    },
+  });
+
+  const resource = db.lookup('resource', 'cloudFormationType', 'equals', 'AWS::OneOf::Required').only();
+  const requiredProps = Object.entries(resource.properties)
+    .filter(([_, value]) => value.required)
+    .map(([name, _]) => name);
+  expect(requiredProps.length).toBe(0);
```

Comment:
Why is it 0 as the expected value? `Type` is required no?

Reply:
`Type` is on the `SomeConfigurationProp` type, it's not a propery of the `AWS::OneOf::Required` resource. 

The only property that could be required is `DataSourceConfiguration`, which is not marked required.